README.md

```diff
@@ -93,10 +83,21 @@ When you're done testing the changes locally, here is how you release a new vers
     1. Using GitHub: Go to the [releases page](/releases) and click "Draft a new release".
     1. Using Git:
 
-    ```
+    ```sh
     git tag -a v0.0.2 -m "tag message"
     git push --follow-tags
     ```
+
 1. Now you can use the new Git tag (e.g. `v0.0.2`) in the `ref` attribute of the `source` URL in `terragrunt.hcl`.
 1. Run `terragrunt plan`.
-1. If the plan looks good, run `terragrunt apply`.   
+1. If the plan looks good, run `terragrunt apply`.
+
+## Monorepo vs. polyrepo
+
+This example shows live modules in a "monorepo" - several modules in a single repository. There are [benefits and drawbacks to using a monorepo](https://github.com/joelparkerhenderson/monorepo_vs_polyrepo) vs. using a "polyrepo" - one module per repository. Which you choose depends on your tooling, how you build/test Terraform modules, and so on.
```

Comment:
One other note: the structure recommended by Terragrunt isn't truly monorepo, as there is a separate infrastructure-live repo. That repo is used to be able to deploy and test different versions of modules in different environments. 

In fact, there are really multiple dimensions here: monorepo vs multiple repos, versioned artifacts vs latest-from-source, and so on. Probably not worth getting into all the details...

Reply:
I'm fine with however you want to phrase it, the larger point I was hoping to get at was:
- There are good technical reasons to keep generic modules one-per-repo. It fits with the Terraform standards, allows you to move to a Terraform registry if you want, and keeps dependencies nicely isolated.
- There are good technical reasons to keep your "live repo" with all the subscriptions/environments/geolocations in a single repo. Ensuring changes across environments are small and controlled, having a 1:1 mapping/overview of what you have deployed, and so on.
- Keeping the [modules | live modules | infrastructure modules] in one repo vs. one-per-repo is more of a judgment call. There are pros and cons to each approach to weigh, and while the example shows these types of modules in a single repo, that's not really part of the guidance the way other things are - pinning Terraform and provider versions, for example.

I'll pick up the pros/cons from your note below.

Reply:
> The phrase "live modules" is a little confusing, as there is a separate infrastructure-live repo. The modules in this repo are just.. modules.

True, but it might be good to call them _something_ other than just "modules." One of the real challenges my team and I have had in distinguishing between "things that are generic and reusable across products/environments" vs. "things that are written specifically for one product/environment" is that they're _all_ "modules." So when someone says, "How do we name modules?" You ask, "Is it a generic module or a... uh... module?" and that gets really confusing.

It seemed to make sense that since the "live repo" is where they're consumed, "live modules" was a reasonable distinction. I'm fine removing it, but it might be good to think of a qualified name for this type of module.

Reply:
> I'm fine with however you want to phrase it, the larger point I was hoping to get at was:

Agreed with all your points!

> True, but it might be good to call them something other than just "modules."

Yea, the term module is very, very overloaded.

I think _root module_ is the official term for a module you are expected to run `apply` in directly. And perhaps we could use _generic module_ to refer to modules meant to be consumed in other modules, but not applied directly.